### PR TITLE
multicluster: support inject / recover on remote cluster

### PR DESCRIPTION
**Pull request by YangKeao**
### What problem does this PR solve?

Inject / recover chaos on remote cluster

### What's changed and how it works?

Documentation: https://github.com/chaos-mesh/website/pull/365.

1. Add a controller on manager cluster to synchronous create new chaos or remove old ones 
2. Add a controller for new cluster to watch the changes on remote cluster

TODO:

1. Synchronize the status
2. Add documents to clarify the workflow of these two controllers.

### CODE REVIEW

File: controllers/multicluster/remotechaos/fx.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+			return err
+		}
+
+	}
+
+	return nil
+}
+
+// this controller will only create or delete the remote chaos
+var remotePredicates = predicate.Funcs{
```
Comment (cwen0) [inline controllers/multicluster/remotechaos/fx.go:85]: Why not using `RemoteChaosPredicate` directly?

File: controllers/multicluster/remotecluster/controller.go [hunk 2]
```diff
@@ -120,3 +152,90 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
...
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+	return nil
+}
+
+func (r *Reconciler) uninstallHelmRelease(ctx context.Context, obj *v1alpha1.RemoteCluster, clientConfig clientcmd.ClientConfig) error {
```
Comment (cwen0) [inline controllers/multicluster/remotecluster/controller.go:215]: Will CRDs be deleted?

**Comment by cwen0** [comment]
@YangKeao CI tests were broken. Can you fix them?

**Review by cwen0 — APPROVED** [review]
LGTM

**Comment by STRRL** [comment]
/hold 

please update the changelog, it should not remove any entries.

**Comment by mdnyeemakhtar** [comment]
Hello @YangKeao,

I'd greatly appreciate if I can get some information on TODO `1. Synchronize the status`. Is it being done in a different PR?
Here is my more detailed question https://github.com/chaos-mesh/chaos-mesh/pull/3396#issuecomment-1474368422

**Pull request merged**